lib/collapsers/binary.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+var logger = require('bole')('collapsify:collapsers:binary');
+var base64Utils = require('../utils/base-64');
+
+function external(url, opts) {
+  logger.info('Fetching external binary from "%s".', url);
+
+  return opts.fetch(url).then(collapse);
```

Comment:
Originally many of the "opts" (such as `resourceLocation`) were passed as parameters, so it wasn't as clear to do it that way.